File: PST/FUEL/src/FUEL/FS/FileUpload.vb
Imports FUEL
Imports FUEL.My
Imports Microsoft.VisualBasic
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.Diagnostics
Imports System.IO
Imports System.Runtime.CompilerServices
Imports System.Threading

Namespace FUEL.FS
    Public Class FileUpload
        ' Methods
        Public Sub New()
            Me._UploadFrequency = 2
            Me._UploadRequired = False
            Me.NewThread = False
            Me.ZipFiles = False
            Me.ZipFileName = Nothing
            Me.DeleteFilesAfterZip = False
            Me.MaxZipFileSize = 0
            Me.HideForms = False
        End Sub

        Public Sub New(ByVal SrcDir As String)
            Me._UploadFrequency = 2
            Me._UploadRequired = False
            Me.NewThread = False
            Me.ZipFiles = False
            Me.ZipFileName = Nothing
            Me.DeleteFilesAfterZip = False
            Me.MaxZipFileSize = 0
            Me.HideForms = False
            Me._SrcDir = SrcDir
            Me.UploadFiles(0)
        End Sub

        Public Sub New(ByVal SrcDir As String, ByVal UploadInterval As Integer)
            Me._UploadFrequency = 2
            Me._UploadRequired = False
            Me.NewThread = False
            Me.ZipFiles = False
            Me.ZipFileName = Nothing
            Me.DeleteFilesAfterZip = False
            Me.MaxZipFileSize = 0
            Me.HideForms = False
            Me._UploadFrequency = UploadInterval
            Me._SrcDir = SrcDir
            Me.UploadFiles(0)
        End Sub

        Public Sub AbortUpload()
            If Me._thdUpload.IsAlive Then
                Me._thdUpload.Abort
            End If
        End Sub

        Public Sub UploadFiles()
            Me.UploadFiles(0)
        End Sub

        <MethodImpl((MethodImplOptions.NoOptimization Or MethodImplOptions.NoInlining))> _
        Private Sub UploadFiles(ByVal Recursion As Integer)
            If Not Me.UploadRequired Then
                Logging.AddLogEntry(Me, "Upload not currently required.", EventLogEntryType.Information, 2)
            Else
                Try 
                    If Not UploadSettings.SettingsVerified Then
                        If (Recursion <> 0) Then
                            Logging.AddLogEntry(Me, "User did not verify settings, skipping request to upload files.", EventLogEntryType.Information, 0)
                            Interaction.MsgBox("Can not upload files until the FUEL Settings have been verified." & ChrW(13) & ChrW(10) & ChrW(13) & ChrW(10) & "Please access FUEL Settings via the Windows Start Menu" & ChrW(13) & ChrW(10) & ChrW(13) & ChrW(10) & "Skipping request to upload files.", MsgBoxStyle.ApplicationModal, Nothing)
                        Else
                            Logging.AddLogEntry(Me, "Cant upload files till FUEL Settings have been verified", EventLogEntryType.Information, 0)
                            Interaction.MsgBox("You must verify FUEL Settings before trying to upload data", MsgBoxStyle.ApplicationModal, Nothing)
                            Dim process As New Process
                            process.StartInfo.FileName = If(Not MyProject.Application.Info.DirectoryPath.Contains("\bin\"), Path.Combine(modCommonCode.GetDataPath, "FuelSettings.exe"), "C:\Users\morrisor\Documents\Visual Studio 2010\Projects\FUEL\AutoSendFiles\FUELSettings\bin\Debug\FUELSettings.exe")
                            process.Start
                            process.WaitForExit
                            Me.UploadFiles((Recursion + 1))
                        End If
                    Else
                        Logging.AddLogEntry("clsFileUpload", "Starting Upload", EventLogEntryType.Information, 2)
                        If Me.NewThread Then
                            Me._thdUpload = New Thread(New ThreadStart(AddressOf Me.UploadInNewThread))
                            Me._thdUpload.IsBackground = False
                            Me._thdUpload.Start
                        Else
                            Dim upload As New frmAutoUpload(Me.SrcDir)
                            Dim aHandle As IntPtr = PST.getOwner
                            If (aHandle <> DirectCast(-1, IntPtr)) Then
                                upload.ShowDialog(New WindowWrapper(aHandle))
                            Else
                                upload.ShowDialog
                            End If
                        End If
                    End If
                    MySettingsProperty.Settings.LastUploadTime = DateAndTime.Now
                    MySettingsProperty.Settings.Save
                Catch exception1 As Exception
                    Dim ex As Exception = exception1
                    ProjectData.SetProjectError(ex)
                    Interaction.MsgBox(("There was a problem uploading your data to the network" & ChrW(13) & ChrW(10) & ChrW(13) & ChrW(10) & ex.ToString), MsgBoxStyle.Critical, Nothing)
                    ProjectData.ClearProjectError
                End Try
            End If
        End Sub

        Private Sub UploadInNewThread()
            Logging.AddLogEntry("FS.UploadInNewThread", "Starting Zip process", EventLogEntryType.Information, 2)
            New frmZipProgress(Me.SrcDir, Me.ZipFileName, Me.DeleteFilesAfterZip) With { _
                .MaxFileSize = CLng(Math.Round(Me.MaxZipFileSize)), _
                .HideForm = Me.HideForms _
            }.ShowDialog
            Logging.AddLogEntry("FS.UploadInNewThread", "Done with Zip process", EventLogEntryType.Information, 2)
            Logging.AddLogEntry("FS.UploadInNewThread", "Starting Upload process", EventLogEntryType.Information, 2)
            Dim upload As New frmAutoUpload(Me.SrcDir) With { _
                .HideForm = Me.HideForms _
            }
            upload.ShowDialog
            Logging.AddLogEntry("FS.UploadInNewThread", "Done with Upload process", EventLogEntryType.Information, 2)
            upload.HideForm = Me.HideForms
        End Sub


        ' Properties
        Private Property _SrcDir As String
            <DebuggerNonUserCode> _
            Get
                Return Me.__SrcDir
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As String)
                Me.__SrcDir = AutoPropertyValue
            End Set
        End Property

        Public Property SrcDir As String
            Get
                Return Me._SrcDir
            End Get
            Set(ByVal value As String)
                Me._SrcDir = value
            End Set
        End Property

        Private Property _UploadFrequency As Integer
            <DebuggerNonUserCode> _
            Get
                Return Me.__UploadFrequency
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As Integer)
                Me.__UploadFrequency = AutoPropertyValue
            End Set
        End Property

        Public Property UploadFrequency As Integer
            Get
                Return Me._UploadFrequency
            End Get
            Set(ByVal value As Integer)
                Me._UploadFrequency = value
            End Set
        End Property

        Private Property _UploadRequired As Boolean
            <DebuggerNonUserCode> _
            Get
                Return Me.__UploadRequired
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As Boolean)
                Me.__UploadRequired = AutoPropertyValue
            End Set
        End Property

        Public ReadOnly Property UploadRequired As Boolean
            Get
                Dim num As Integer = CInt(DateAndTime.DateDiff(DateInterval.Minute, MySettingsProperty.Settings.LastUploadTime, DateAndTime.Now, FirstDayOfWeek.Sunday, FirstWeekOfYear.Jan1))
                Logging.AddLogEntry("FileUpload: UploadRequired", ("Date Delta from last upload (minutes) = " & num.ToString), EventLogEntryType.Information, 2)
                Return (num >= (Me._UploadFrequency * 60))
            End Get
        End Property

        Public Property NewThread As Boolean
            <DebuggerNonUserCode> _
            Get
                Return Me._NewThread
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As Boolean)
                Me._NewThread = AutoPropertyValue
            End Set
        End Property

        Public Property ZipFiles As Boolean
            <DebuggerNonUserCode> _
            Get
                Return Me._ZipFiles
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As Boolean)
                Me._ZipFiles = AutoPropertyValue
            End Set
        End Property

        Public Property ZipFileName As String
            <DebuggerNonUserCode> _
            Get
                Return Me._ZipFileName
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As String)
                Me._ZipFileName = AutoPropertyValue
            End Set
        End Property

        Public Property DeleteFilesAfterZip As Boolean
            <DebuggerNonUserCode> _
            Get
                Return Me._DeleteFilesAfterZip
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As Boolean)
                Me._DeleteFilesAfterZip = AutoPropertyValue
            End Set
        End Property

        Public Property MaxZipFileSize As Double
            <DebuggerNonUserCode> _
            Get
                Return Me._MaxZipFileSize
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As Double)
                Me._MaxZipFileSize = AutoPropertyValue
            End Set
        End Property

        Public Property HideForms As Boolean
            <DebuggerNonUserCode> _
            Get
                Return Me._HideForms
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As Boolean)
                Me._HideForms = AutoPropertyValue
            End Set
        End Property

        Public ReadOnly Property Status As String
            Get
                Return If(Not Me._thdUpload.IsAlive, "Idle", "Busy")
            End Get
        End Property


        ' Fields
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private __SrcDir As String
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private __UploadFrequency As Integer
        <CompilerGenerated, DebuggerBrowsable(DebuggerBrowsableState.Never)> _
        Private __UploadRequired As Boolean
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private _NewThread As Boolean
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private _ZipFiles As Boolean
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private _ZipFileName As String
        <CompilerGenerated, DebuggerBrowsable(DebuggerBrowsableState.Never)> _
        Private _DeleteFilesAfterZip As Boolean
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private _MaxZipFileSize As Double
        <CompilerGenerated, DebuggerBrowsable(DebuggerBrowsableState.Never)> _
        Private _HideForms As Boolean
        Private _thdUpload As Thread
    End Class
End Namespace

File: PST/FUEL Source Code/FUEL/Backup/FS/FileUpload.cs
// Decompiled with JetBrains decompiler
// Type: FUEL.FS.FileUpload
// Assembly: FUEL, Version=2.4.8.0, Culture=neutral, PublicKeyToken=null
// MVID: A3D630D0-0620-425D-8CFB-B67B2AC75507
// Assembly location: C:\hp\FUEL\FUEL.dll

using FUEL.My;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using System;
using System.Diagnostics;
using System.IO;
using System.Runtime.CompilerServices;
using System.Threading;
using System.Windows.Forms;

namespace FUEL.FS
{
  public class FileUpload
  {
    private Thread _thdUpload;

    private string _SrcDir { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    public string SrcDir
    {
      get => this._SrcDir;
      set => this._SrcDir = value;
    }

    private int _UploadFrequency { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    public int UploadFrequency
    {
      get => this._UploadFrequency;
      set => this._UploadFrequency = value;
    }

    private bool _UploadRequired { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    public bool UploadRequired
    {
      get
      {
        int num = checked ((int) DateAndTime.DateDiff(DateInterval.Minute, MySettingsProperty.Settings.LastUploadTime, DateAndTime.Now));
        Logging.AddLogEntry((object) "FileUpload: UploadRequired", "Date Delta from last upload (minutes) = " + num.ToString(), EventLogEntryType.Information, 2);
        return num >= checked (this._UploadFrequency * 60);
      }
    }

    public bool NewThread { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    public bool ZipFiles { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    public string ZipFileName { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    public bool DeleteFilesAfterZip { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    public double MaxZipFileSize { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    public bool HideForms { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    public FileUpload()
    {
      this._UploadFrequency = 2;
      this._UploadRequired = false;
      this.NewThread = false;
      this.ZipFiles = false;
      this.ZipFileName = (string) null;
      this.DeleteFilesAfterZip = false;
      this.MaxZipFileSize = 0.0;
      this.HideForms = false;
    }

    public FileUpload(string SrcDir)
    {
      this._UploadFrequency = 2;
      this._UploadRequired = false;
      this.NewThread = false;
      this.ZipFiles = false;
      this.ZipFileName = (string) null;
      this.DeleteFilesAfterZip = false;
      this.MaxZipFileSize = 0.0;
      this.HideForms = false;
      this._SrcDir = SrcDir;
      this.UploadFiles(0);
    }

    public FileUpload(string SrcDir, int UploadInterval)
    {
      this._UploadFrequency = 2;
      this._UploadRequired = false;
      this.NewThread = false;
      this.ZipFiles = false;
      this.ZipFileName = (string) null;
      this.DeleteFilesAfterZip = false;
      this.MaxZipFileSize = 0.0;
      this.HideForms = false;
      this._UploadFrequency = UploadInterval;
      this._SrcDir = SrcDir;
      this.UploadFiles(0);
    }

    public void UploadFiles() => this.UploadFiles(0);

    [MethodImpl(MethodImplOptions.NoInlining | MethodImplOptions.NoOptimization)]
    private void UploadFiles(int Recursion)
    {
      if (this.UploadRequired)
      {
        try
        {
          if (UploadSettings.SettingsVerified)
          {
            Logging.AddLogEntry((object) "clsFileUpload", "Starting Upload", EventLogEntryType.Information, 2);
            if (!this.NewThread)
            {
              frmAutoUpload frmAutoUpload = new frmAutoUpload(this.SrcDir);
              IntPtr owner = PST.getOwner();
              if (owner != (IntPtr) -1)
              {
                int num1 = (int) frmAutoUpload.ShowDialog((IWin32Window) new WindowWrapper(owner));
              }
              else
              {
                int num2 = (int) frmAutoUpload.ShowDialog();
              }
            }
            else
            {
              this._thdUpload = new Thread(new ThreadStart(this.UploadInNewThread));
              this._thdUpload.IsBackground = false;
              this._thdUpload.Start();
            }
          }
          else if (Recursion == 0)
          {
            Logging.AddLogEntry((object) this, "Cant upload files till FUEL Settings have been verified", EventLogEntryType.Information, 0);
            int num = (int) Interaction.MsgBox((object) "You must verify FUEL Settings before trying to upload data");
            Process process = new Process();
            process.StartInfo.FileName = !MyProject.Application.Info.DirectoryPath.Contains("\\bin\\") ? Path.Combine(modCommonCode.GetDataPath(), "FuelSettings.exe") : "C:\\Users\\morrisor\\Documents\\Visual Studio 2010\\Projects\\FUEL\\AutoSendFiles\\FUELSettings\\bin\\Debug\\FUELSettings.exe";
            process.Start();
            process.WaitForExit();
            this.UploadFiles(checked (Recursion + 1));
          }
          else
          {
            Logging.AddLogEntry((object) this, "User did not verify settings, skipping request to upload files.", EventLogEntryType.Information, 0);
            int num = (int) Interaction.MsgBox((object) "Can not upload files until the FUEL Settings have been verified.\r\n\r\nPlease access FUEL Settings via the Windows Start Menu\r\n\r\nSkipping request to upload files.");
          }
          MySettingsProperty.Settings.LastUploadTime = DateAndTime.Now;
          MySettingsProperty.Settings.Save();
        }
        catch (Exception ex)
        {
          ProjectData.SetProjectError(ex);
          int num = (int) Interaction.MsgBox((object) ("There was a problem uploading your data to the network" + "\r\n\r\n" + ex.ToString()), MsgBoxStyle.Critical);
          ProjectData.ClearProjectError();
        }
      }
      else
        Logging.AddLogEntry((object) this, "Upload not currently required.", EventLogEntryType.Information, 2);
    }

    private void UploadInNewThread()
    {
      Logging.AddLogEntry((object) "FS.UploadInNewThread", "Starting Zip process", EventLogEntryType.Information, 2);
      int num1 = (int) new frmZipProgress(this.SrcDir, this.ZipFileName, this.DeleteFilesAfterZip)
      {
        MaxFileSize = (checked ((long) System.Math.Round(this.MaxZipFileSize))),
        HideForm = this.HideForms
      }.ShowDialog();
      Logging.AddLogEntry((object) "FS.UploadInNewThread", "Done with Zip process", EventLogEntryType.Information, 2);
      Logging.AddLogEntry((object) "FS.UploadInNewThread", "Starting Upload process", EventLogEntryType.Information, 2);
      frmAutoUpload frmAutoUpload = new frmAutoUpload(this.SrcDir);
      frmAutoUpload.HideForm = this.HideForms;
      int num2 = (int) frmAutoUpload.ShowDialog();
      Logging.AddLogEntry((object) "FS.UploadInNewThread", "Done with Upload process", EventLogEntryType.Information, 2);
      frmAutoUpload.HideForm = this.HideForms;
    }

    public string Status => this._thdUpload.IsAlive ? "Busy" : "Idle";

    public void AbortUpload()
    {
      if (!this._thdUpload.IsAlive)
        return;
      this._thdUpload.Abort();
    }
  }
}
